scripts/add_existing_baseyear.py

```diff
@@ -0,0 +1,394 @@
+
+    df_agg.drop(df_agg.index[df_agg.Fueltype.isin(fueltype_to_drop)],inplace=True)
+    df_agg.drop(df_agg.index[df_agg.Technology.isin(technology_to_drop)],inplace=True)
+    df_agg.Fueltype = df_agg.Fueltype.map(rename_fuel)
+
+    # add existing solar and wind capacities
+    # source: https://www.irena.org/Statistics/Download-Data
+    cc = pd.read_csv('data/Country_codes.csv', sep=',', index_col=-1)
+    name_to_2code = dict(zip(cc['Country'].tolist(),
+                             cc['2 letter code (ISO-3166-2)'].tolist()))
+    for tech in ['solar', 'onwind', 'offwind']:
+        df = pd.read_csv('data/existing_infrastructure/{}_capacity_IRENA.csv'.format(tech), 
+                         sep=',', index_col=0)
+
+        df.rename(index={'Czechia':'Czech Republic',
+                         'UK':'United Kingdom',
+                         'Bosnia Herzg':'Bosnia Herzegovina',
+                         'North Macedonia': 'Macedonia'}, inplace=True)
+
+        df.rename(index=lambda country : name_to_2code[country], inplace=True)
+
+        # calculate yearly differences
+        df.insert(loc=0, value=.0, column='1999')
+        df = df.diff(axis=1).drop('1999', axis=1)
+        df = df.clip(lower=0)
+        df.replace(to_replace=0.0, value=np.nan, inplace=True)
+
+        for year in df.columns:
+            for country in df.index:
+                if df.notnull().loc[country,year]:
+                    df_agg = df_agg.append({'Fueltype':tech,
+                                            'Country':country,
+                                            'Capacity':df.loc[country,year],
+                                            'YearCommissioned':int(year),
+                                            'YearDecommissioning':int(float(year)+costs.at[tech, 'lifetime'])},
+                                             ignore_index=True)
+           
+        # powerplants already include an estimated decommissining year wich has been
+        # calculated based on pm config, check if we want to make this calculation more
+        # transparent, e.g., by loading lifetime assumptions from costs.csv into pm config
+        # check how pm is dealing with retrofitting
+        # pm.get_config()['fuel_to_lifetime']
+        # index = df_agg.Fueltype.map(lf)+df_agg.YearCommissioned > 2020
+
+        index=df_agg.YearDecommissioning > int(year)
+        df = df_agg[index].pivot_table(index='Country', columns='Fueltype',
+                                       values='Capacity', aggfunc='sum')
+
+        nodes=set([node[0:2] for node in n.buses.index[n.buses.carrier == "AC"]])
+
+    for carrier in ['coal', 'oil', 'uranium']:
+        n.add("Bus",
+              "EU " + carrier,
+              carrier=carrier)
+    
+    for node in nodes:
+        #if a country has more than one node, selects the first one
+        bus_selected=[bus for bus in n.buses.index[n.buses.carrier == "AC"] if bus[0:2]==node][0]  
```

Comment:
We need to fix this soonish so that it attaches generators to the correct node, or it should go on a TODO list. Powerplantmatching has the coordinates.

Reply:
For those plants included in powerplantmatching should be easy, for existing wind and solar capacities we need to decide how to split among nodes in a country